src/pages/iv-items/forms/BarcodeForm.js

```diff
@@ -0,0 +1,171 @@
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useContext, useEffect, useState } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { useForm } from 'src/hooks/form'
+
+const BarcodeForm = ({ store, labels, maxAccess }) => {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const [numRows, setNumRows] = useState(0)
+
+  const { formik } = useForm({
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      barcodes: yup
+        .array()
+        .of(
+          yup.object().shape({
+            barcode: yup.string().test(function (value) {
+              if (numRows > 1) {
+                return !!value
+              }
+
+              return true
+            })
+          })
+        )
+        .required()
+    }),
+    initialValues: {
+      barcodes: [
+        {
+          id: 1,
+          itemId: recordId,
+          barcode: '',
+          muId: '',
+          defaultQty: '',
+          muRef: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      const { barcodes } = values
+
+      const isRowEmpty = row => {
+        return !row.barcode
+      }
+
+      if (barcodes.length === 1 && isRowEmpty(barcodes[0])) {
+        formik.setValues({ barcodes: [] })
+        postdata([])
+      } else {
+        postdata(values)
+      }
+    }
+  })
+
+  const postdata = obj => {
+    const data = obj?.barcodes?.map(({ itemId, ...rest }) => ({
+      itemId: recordId,
+      ...rest
+    }))
+
+    const list = {
+      itemId: recordId,
+      barcodes: data || []
+    }
+
+    postRequest({
+      extension: InventoryRepository.Barcode.set2,
+      record: JSON.stringify(list)
+    })
+      .then(res => {
+        toast.success(platformLabels.Edited)
+        getData()
+      })
+      .catch(error => {})
```

Comment:
if there is no try{ why would you need a catch?
please do not add the try just remove the unused catch
api errors are now handled in the RequestsContext